src/kakarot/instructions.cairo

```diff
@@ -611,6 +611,7 @@ namespace EVMInstructions {
         // Terminate execution
         if (stopped != FALSE) {
             if (is_parent_root != FALSE) {
+                let ctx = SelfDestructHelper.selfdestruct(ctx);
```

Comment:
I think we should call selfdestruct only if the destroy_contracts_len is not 0, it feels risky when reading the whole codeflow that the selfdestruct() is called everytime